src/web/signer.ts

```diff
@@ -57,8 +59,7 @@ export class TurboWebArweaveSigner implements TurboWalletSigner {
       size: fileSizeFactory(),
     });
     this.logger.debug('Signing data item...');
-    // TODO: support target, anchor and tags for upload
-    const signedDataItem = createData(buffer, this.signer, {});
+    const signedDataItem = createData(buffer, this.signer, opts);
```

Comment:
May want write a utility to assert that the opts are healthy, e.g. the following criteria:

- https://github.com/Irys-xyz/arbundles/blob/master/src/ar-data-create.ts#L64
- https://github.com/Irys-xyz/arbundles/blob/master/src/ar-data-create.ts#L53
- https://github.com/ardriveapp/upload-service/blob/dev/src/bundles/streamingDataItem.ts#L273-L274
- https://github.com/ardriveapp/upload-service/blob/dev/src/bundles/streamingDataItem.ts#L317-L354

Reply:
in the current iteration we will run the arbundles assertions there during data item creation and then send to upload service and upload service will run those assertions. so in a way we're covered there already